packages/wallets/src/features/accounts/screens/IDVDocumentUpload/components/IDVDocumentUploadDetails/IDVDocumentUploadDetails.tsx

```diff
@@ -7,16 +9,46 @@ import unixToDateString from '../../../../../../utils/utils';
 import { dateOfBirthValidator, firstNameValidator, lastNameValidator } from '../../../../validations';
 import './IDVDocumentUploadDetails.scss';
 
-const IDVDocumentUploadDetails = () => {
+type TIdvDocumentUploadDetailsProps = {
+    isDetailsVerified: boolean;
+    setIsDetailsVerified: React.Dispatch<React.SetStateAction<boolean>>;
+};
+
+const IDVDocumentUploadDetails: React.FC<TIdvDocumentUploadDetailsProps> = ({
+    isDetailsVerified,
+    setIsDetailsVerified,
+}) => {
     const { data: getSettings } = useSettings();
-    const { setFormValues } = useFlow();
+    const { errors, formValues, setFormValues } = useFlow();
+    const { validateForm } = useFormikContext();
 
     const handleDateChange = (formattedDate: string | null) => {
         setFormValues('dateOfBirth', formattedDate);
     };
 
     const dateOfBirth = getSettings?.date_of_birth || 0;
     const formattedDateOfBirth = new Date(dateOfBirth * 1000);
+    const firstName = getSettings?.first_name;
+    const lastName = getSettings?.last_name;
+
+    useEffect(() => {
+        setFormValues('firstName', getSettings?.first_name);
+        setFormValues('lastName', getSettings?.last_name);
+        validateForm();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [getSettings?.first_name, getSettings?.last_name]);
```

Comment:
Updates the form values of `firstName` and `lastName` on initial render and on every update of the values in getSettings to prevent checkbox from being disabled when formValues is not immediately updated with the getSettings values